Makefile

```diff
@@ -194,32 +193,64 @@ $(TERRAFORM):
 	@rm -fr $(TOOLS_HOST_DIR)/tmp-terraform
 	@$(OK) installing terraform $(HOSTOS)-$(HOSTARCH)
 
-$(TERRAFORM_PROVIDER):
-	@$(INFO) installing terraform AWS provider for the platform $(SAFEHOST_PLATFORM)
-	@mkdir -p $(dir $(TERRAFORM_PROVIDER))
-	@curl -fsSL https://github.com/upbound/terraform-provider-aws/releases/download/$(TERRAFORM_PROVIDER_RELEASE)/terraform-provider-aws_$(TERRAFORM_PROVIDER_RELEASE)_$(SAFEHOST_PLATFORM).zip -o $(TERRAFORM_PROVIDER).zip
-	@unzip $(TERRAFORM_PROVIDER).zip -d $(dir $(TERRAFORM_PROVIDER))
-	@rm -fr $(TERRAFORM_PROVIDER).zip
-	@$(OK) installing terraform provider $(SAFEHOST_PLATFORM)
+$(TERRAFORM_PROVIDER): $(UPSTREAM)
+	@$(INFO) Building terraform AWS provider from source if changes detected
+	@(cd $(ROOT_DIR)/upstream && make go-build)
+	@$(OK) Built terraform AWS provider from source
 
+# TODO(mbbush) Remove this when we can. This is the last thing left over from when we used to fork a terraform cli
+# process for each resource. This generates the tf provider's json schema, which is slightly different from the go
+# schema (float vs int for some types). To avoid many schema changes, we added code in config/registry.go to use the
+# json schema for schema generation. Now that we have multi-version CRDs, it's easier to manage the schema change.
 $(TERRAFORM_PROVIDER_SCHEMA): $(TERRAFORM) $(TERRAFORM_PROVIDER)
-	@$(INFO) generating provider schema for $(TERRAFORM_PROVIDER_SOURCE) $(TERRAFORM_PROVIDER_VERSION)
+	@$(INFO) generating provider schema for $(TERRAFORM_PROVIDER_SOURCE) from the upstream git submodule
 	@rm -fr $(TERRAFORM_WORKDIR)
 	@mkdir -p $(TERRAFORM_WORKDIR)
-	@echo '{"terraform":[{"required_providers":[{"provider":{"source":"'"$(TERRAFORM_PROVIDER_SOURCE)"'","version":"'"$(TERRAFORM_PROVIDER_VERSION)"'"}}],"required_version":"'"$(TERRAFORM_VERSION)"'"}]}' > $(TERRAFORM_WORKDIR)/main.tf.json
-	@echo 'provider_installation { filesystem_mirror { path = "'"$(TERRAFORM_FILESYSTEM_MIRROR)"'", include = ["hashicorp/aws"] }, direct { exclude = ["hashicorp/aws"] } }' > $(TERRAFORM_CLI_CONFIG_FILE)
-	@TF_CLI_CONFIG_FILE=$(TERRAFORM_CLI_CONFIG_FILE) $(TERRAFORM) -chdir=$(TERRAFORM_WORKDIR) init -upgrade > $(TERRAFORM_WORKDIR)/terraform-logs.txt 2>&1
-	@TF_CLI_CONFIG_FILE=$(TERRAFORM_CLI_CONFIG_FILE) $(TERRAFORM) -chdir=$(TERRAFORM_WORKDIR) providers schema -json=true > $(TERRAFORM_PROVIDER_SCHEMA) 2>> $(TERRAFORM_WORKDIR)/terraform-logs.txt
-	@$(OK) generating provider schema for $(TERRAFORM_PROVIDER_SOURCE) $(TERRAFORM_PROVIDER_VERSION)
+	@echo 'data "aws_partition" "example" {}' > $(TERRAFORM_WORKDIR)/main.tf
```

Comment:
Is this arbitrary? Why don't we use the old one? When we consider together with [the comment](https://github.com/crossplane-contrib/provider-upjet-aws/pull/1735/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R184), then will versioning work?

Reply:
I basically replaced the process we had for generating the schema.json file with the one the terraform provider itself had for generating the same file. I didn't want to try to implement a new build process for the upstream provider, so I was trying to keep it simple. 

As I think about it again, it seems like it might be possible for my simple workflow to download a provider from the terraform registry, which is not what we want, so I think I'll incorporate more of what was there before. To the extent I can, I'll avoid aws-specific things, so that as much as possible can be reused with the other providers. Thanks for the callout.